src/screens/PostScreen.tsx

```diff
@@ -1,43 +1,117 @@
-        dedicated to addressing both immediate needs and underlying causes of
-        homelessness through a combination of shelter services, job training
-        programs, and mental health support. read more online:
-        https://www.codetenderloin.org/
-      </Text>
-      <Text>
-        Image URL:
-        'https://cdn.britannica.com/51/178051-050-3B786A55/San-Francisco.jpg'
-      </Text>
-      <HeartIcon />
-      <Text>256 Likes</Text>
-      <ShareIcon />
-      <ProfilePlaceholder />
-      <Text>philip_ye</Text>
-      <Text>September 20</Text>
-      <Text>
-        This organization is doing amazing work tackling the complex root causes
-        of the issue.
-      </Text>
-      <ProfilePlaceholder />
-      <Text>vppraggie</Text>
-      <Text>September 21</Text>
-      <Text>Thanks for sharing!</Text>
+      {postData.map(post => (
+        <View key={post.id}>
+          {/* Profile Header */}
+          <View style={styles.profileHeader}>
+            <View style={{ flexDirection: 'row', alignItems: 'center' }}>
+              <ProfilePlaceholder style={styles.profileIcon} />
+              <Text style={styles.profileName}>{post.username}</Text>
+            </View>
+            <Text style={styles.postDate}>
+              {new Date(post.created_at).toLocaleDateString('en-US', {
+                year: 'numeric',
+                month: 'long',
+                day: 'numeric',
+              })}
+            </Text>
+          </View>
+
+          {/* Post Text */}
+          <View style={styles.postTextContainer}>
+            <Text style={styles.postText}>{post.description}</Text>
+          </View>
+
+          {/* Post Image */}
+          {post.image_url ? (
+            <Image source={{ uri: post.image_url }} style={styles.postImage} />
+          ) : null}
+
+          {/* Interaction Row */}
+          <View style={styles.interactionRow}>
+            <View style={{ flexDirection: 'row', alignItems: 'center' }}>
+              <HeartIcon style={styles.icon} />
+              <Text style={styles.likesText}>{post.like_count} Likes</Text>
+            </View>
+            <ShareIcon style={styles.icon} />
+          </View>
+
+          {/* Comment Section */}
+          <View style={styles.commentSection}>
+            {/* First Comment */}
+            <View style={styles.comment}>
+              <ProfilePlaceholder style={styles.commentIcon} />
+              <View style={styles.commentBody}>
+                <View style={styles.commentHeader}>
+                  <Text style={styles.commentName}>philip_ye</Text>
+                  <Text style={styles.commentDate}>September 20</Text>
+                </View>
+                <Text style={styles.commentText}>
+                  This organization is doing amazing work tackling the complex
+                  root causes of the issue.
+                </Text>
+              </View>
+            </View>
```

Comment:
Given that there are so many divs for each div, might be helpful to create a separate `Comment` component.

There's also a comments table that you can pull this data from!